staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1194,7 +1183,9 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 			defer wg.Done()
 
 			for index := range toProcess {
+				itemsLock.RLock()
 				accessor, err := meta.Accessor(items[index])
+				itemsLock.RUnlock()
```

Comment:
any reason not to just use a channel with capacity and get rid of the locking?

Reply:
Good call - fixed in the second commit.